server/schedule/operator_controller.go

```diff
@@ -301,19 +335,40 @@ func (oc *OperatorController) PromoteWaitingOperator() {
 // - There is no such region in the cluster
 // - The epoch of the operator and the epoch of the corresponding region are no longer consistent.
 // - The region already has a higher priority or same priority operator.
+// - At least one operator is expired.
 func (oc *OperatorController) checkAddOperator(ops ...*operator.Operator) bool {
 	for _, op := range ops {
 		region := oc.cluster.GetRegion(op.RegionID())
 		if region == nil {
-			log.Debug("region not found, cancel add operator", zap.Uint64("region-id", op.RegionID()))
+			log.Debug("region not found, cancel add operator",
+				zap.Uint64("region-id", op.RegionID()))
 			return false
 		}
-		if region.GetRegionEpoch().GetVersion() != op.RegionEpoch().GetVersion() || region.GetRegionEpoch().GetConfVer() != op.RegionEpoch().GetConfVer() {
-			log.Debug("region epoch not match, cancel add operator", zap.Uint64("region-id", op.RegionID()), zap.Reflect("old", region.GetRegionEpoch()), zap.Reflect("new", op.RegionEpoch()))
+		if region.GetRegionEpoch().GetVersion() != op.RegionEpoch().GetVersion() ||
+			region.GetRegionEpoch().GetConfVer() != op.RegionEpoch().GetConfVer() {
+			log.Debug("region epoch not match, cancel add operator",
+				zap.Uint64("region-id", op.RegionID()),
+				zap.Reflect("old", region.GetRegionEpoch()),
+				zap.Reflect("new", op.RegionEpoch()))
 			return false
 		}
 		if old := oc.operators[op.RegionID()]; old != nil && !isHigherPriorityOperator(op, old) {
-			log.Debug("already have operator, cancel add operator", zap.Uint64("region-id", op.RegionID()), zap.Reflect("old", old))
+			log.Debug("already have operator, cancel add operator",
+				zap.Uint64("region-id", op.RegionID()),
+				zap.Reflect("old", old))
+			return false
+		}
+		if op.CheckExpired() {
```

Comment:
There are two operators if create merge operators, maybe only the first one change to expired status.

Reply:
Fixed